File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/VoitureListDto.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;
import org.autostock.models.User;

import java.math.BigDecimal;

@Getter
@Setter
public class VoitureListDto {
    private Long id;
    private String marque;
    private String modele;
    private Integer annee;
    private String couleur;
    private String vin;
    private BigDecimal prixVente;
    private Long owner;
    private String statut; // "EN_STOCK", "VENDUE", etc.

    // getters / setters
}

File: autos-stock-client/src/app/models/VoitureListDto.ts
import {StatutVoiture} from "./enums/StatutVoiture";

export interface VoitureListDto {
  id: number;
  marque: string;
  modele: string;
  annee: number;
  prixDemande: number;
  prixVente: number;
  statut: StatutVoiture;
  needsRemark: boolean;
}
